app/components/record_actions/view.rb

```diff
@@ -45,5 +55,17 @@ def withdraw_link
     def reinstate_link
       govuk_link_to(t("views.trainees.edit.reinstate"), trainee_reinstatement_path(trainee), class: "reinstate")
     end
+
+    def trainee_commencement_date_in_future?
+      trainee.commencement_date && trainee.commencement_date > Time.zone.now
```

Comment:
I think this will need to consider the `trainee.course_start_date`. They can only delete if the `trainee.course_start_date` is in the future and `trainee.commencement_date` is `nil` as a result of them selecting the third option here. If they select either of the first two options, then it's safe to assume the `trainee.course_start_date` is either on the same day or in the past and they cannot delete the trainee
<img width="843" alt="Screenshot 2021-11-09 at 12 55 00" src="https://user-images.githubusercontent.com/616080/140928016-a55ce582-a98f-4e5a-8a5a-8a7a440ebfcb.png">
e